source/features/conversation-navigation.tsx

```diff
@@ -0,0 +1,179 @@
+			// Convert cursor into a number to make operations easier
+			cursor: Number.parseInt(atob(edge.cursor).replace('cursor:', ''), 10)
+		})));
+	},
+	{
+		cacheKey: ([listQuery]) => `${__filebasename}:${JSON.stringify(listQuery)}`,
+		maxAge: 1 / 24 // 1 hour from a day is enough
+	}
+);
+
+// Current default number of items in a conversation list
+const ITEMS_PER_PAGE = 25;
+
+async function getConversationList() {
+	const listQuery = getListQuery();
+	const list = await fetchConversationList(listQuery);
+	const currentNumber = getConversationNumber();
+	const conversation = list.find(item => item.number === currentNumber);
+	const after = (listQuery.page - 1) * ITEMS_PER_PAGE;
+	const before = (listQuery.page * ITEMS_PER_PAGE) + 1;
+
+	if (conversation!.cursor !== after && conversation!.cursor !== before) {
+		return {
+			list,
+			listQuery
+		};
+	}
+
+	if (conversation!.cursor === after) {
+		// `currentConversation` is from previous page, we need to fetch previous page items
+		listQuery.page -= 1;
+	} else if (conversation!.cursor === before) {
+		// `currentConversation` is from next page, we need to fetch next page items
+		listQuery.page += 1;
+	}
+
+	return {
+		list: await fetchConversationList(listQuery),
+		listQuery
+	};
+}
+
+const noListQuery = (): boolean => !(
+	// Allowed also from global conversation list
+	pageDetect.isConversationList(new URL(document.referrer)) ||
+	(new URL(location.href)).searchParams.has('q')
+);
+
+function getListQuery() {
+	const referrerUrl = new URL(document.referrer);
+	const url = pageDetect.isConversationList(referrerUrl) ? referrerUrl : new URL(location.href);
+	const searchQuery = new SearchQuery(url);
+
+	if (pageDetect.isRepoConversationList(referrerUrl)) {
+		searchQuery.add(`repo:${getRepoURL()}`);
+	}
+
+	return {
+		page: Number.parseInt(url.searchParams.get('page') ?? '1', 10),
+		query: searchQuery.get()
+	};
+}
+
+// When fetching a page of conversations we want to also fetch last item from previous page, and first item from next page.
+// This is needed for cases when conversation is first/last in list, and we need to know if items exist on previous/next page.
+// To take into consideration also, than when we are on first item on first page, we don't have a previous item, and we don't want to fetch using a negative cursor.
+function getConversationSearchQuery({page, query}: ReturnType<typeof getListQuery>): string {
+	return `
+search(
+	type: ISSUE,
+	first: ${page === 1 ? ITEMS_PER_PAGE + 1 : ITEMS_PER_PAGE + 2},
+	after: "${btoa(`cursor:${page === 1 ? 0 : ((page - 1) * ITEMS_PER_PAGE) - 1}`)}",
+	query: "${query}"
+) {
+	edges {
+		cursor
+		node {
+			... on Issue { number, url }
+			... on PullRequest { number, url }
+		}
+	}
+}
+	`;
+}
+
+function getConversationNumber() {
+	const conversationHashtag = select('.gh-header-number')?.textContent ?? '#';
+	return Number.parseInt(conversationHashtag.replace('#', ''), 10);
```

Comment:
```suggestion
	return looseParseInt(conversationHashtag);
```

I think this should work